### ISSUE TITLE
no-null: false positive when null is used as function argument

### ISSUE THREAD

**Issue by kkmuffme** [sindresorhus/eslint-plugin-unicorn#1842]
eslint unicorn v42.0.0

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md

Fails when null is used as function argument, even though this should pass. In fact this is even an example given in the "Pass" section:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md#pass

This works (no error):

```js
const foo = Object.create(null);
```

but this fails:

```js
drawingManager.setMap( null );

markers[ index ].setMap( null );
```

I think this shouldn't fail? I can't choose what argument types a function accepts in many cases (libraries). And if we want to have this error for functions we can control, it's better to give an error for no null arg when function is declared?

**Comment by fisker** [comment]
I guess we can add an option for this.

**Comment by ardunster** [comment]
Running into the same issue making MSW mocks with `HttpResponse`, since the first parameter to `HttpResponse`'s constructor is typed as `body?: BodyInit | null` and I need to mock a `null` case.
